.gitattributes

```diff
@@ -38,68 +38,81 @@
+runtime/i386.S      typo.long-line
 
-testsuite/tests/**                                      ocaml-typo=missing-header
-testsuite/tests/lib-bigarray-2/bigarrf.f                ocaml-typo=missing-header,tab
-testsuite/tests/lib-unix/win-stat/fakeclock.c           ocaml-typo=
-testsuite/tests/misc-unsafe/almabench.ml                ocaml-typo=missing-header,long-line
-testsuite/tests/tool-toplevel/strings.ml                ocaml-typo=utf8,missing-header
-testsuite/tests/typing-unboxed-types/test.ml.reference* ocaml-typo=white-at-eof,missing-header
-testsuite/tests/win-unicode/*.ml                        ocaml-typo=utf8,missing-header
-testsuite/tools/*.S                                     ocaml-typo=missing-header
-testsuite/tools/*.asm                                   ocaml-typo=missing-header
-testsuite/typing                                        ocaml-typo=missing-header
+stdlib/hashbang     typo.white-at-eol typo.missing-lf
+
+testsuite/tests/**                                      typo.missing-header
+testsuite/tests/lib-bigarray-2/bigarrf.f                typo.tab
+testsuite/tests/lib-unix/win-stat/fakeclock.c           typo.missing-header=false
+testsuite/tests/misc-unsafe/almabench.ml                typo.long-line
+testsuite/tests/tool-toplevel/strings.ml                typo.utf8
+testsuite/tests/typing-unboxed-types/test.ml.reference* typo.white-at-eof
+testsuite/tests/win-unicode/*.ml                        typo.utf8
+testsuite/tools/*.S                                     typo.missing-header
+testsuite/tools/*.asm                                   typo.missing-header
+testsuite/typing                                        typo.missing-header
+
+# prune testsuite reference files
+testsuite/tests/**/*.reference               typo.prune
+testsuite/tests/**/reference                 typo.prune
 
 # Expect tests with overly long lines of expected output
-testsuite/tests/parsing/docstrings.ml        ocaml-typo=long-line,very-long-line,missing-header
-testsuite/tests/typing-objects/Exemples.ml   ocaml-typo=long-line,missing-header
-testsuite/tests/typing-modules/firstclass.ml ocaml-typo=long-line,missing-header
-testsuite/tests/typing-gadts/test.ml         ocaml-typo=long-line,missing-header
-testsuite/tests/typing-gadts/pr7160.ml       ocaml-typo=long-line,missing-header
+testsuite/tests/parsing/docstrings.ml        typo.long-line typo.very-long-line
+testsuite/tests/typing-objects/Exemples.ml   typo.long-line
+testsuite/tests/typing-modules/firstclass.ml typo.long-line
+testsuite/tests/typing-gadts/test.ml         typo.long-line
+testsuite/tests/typing-gadts/pr7160.ml       typo.long-line
+testsuite/tests/typing-modules/pr7726.ml     typo.long-line
 
-tools/magic         ocaml-typo=missing-header
+tools/magic         typo.missing-header
 
-yacc/*.[ch]   ocaml-typo=long-line,very-long-line,unused-prop
+yacc/*.[ch]         typo.long-line typo.very-long-line=may
 
 # Line-ending specifications, for Windows interoperability
```

Comment:
I like the separation of lint attributes and line-ending controls. Given the overall changes to this file, can we restructure so that all lint instructions (except, perhaps, the one for `.gitattributes` as a very special case) are at the bottom. So can this point onwards be moved below the `Changes` merge driver comment on line 38?